src/WorldMap.cpp

```diff
@@ -1,33 +1,228 @@
+		if (col == m_tileSheetDetails.m_columns)
+		{
+			col = 0;
+			++row;
+		}
+	}
+
+	//Get location on the tile sheet
+	const int tileSize = m_tileSheetDetails.m_tileSize;
+	const int margin = m_tileSheetDetails.m_margin;
+	const int spacing = m_tileSheetDetails.m_spacing;
+
+	return sf::IntRect((margin + ((spacing + tileSize) * col)),
+		(margin + ((spacing + tileSize) * row)),
+		tileSize, tileSize);
+}
+
+sf::IntRect WorldMap::TileSheet::getTileLocationByPosition(const sf::IntRect& position) const
+{
+	const int tileSize = m_tileSheetDetails.m_tileSize;
+	return sf::IntRect(position.left * tileSize, position.top * tileSize, position.width * tileSize, position.height * tileSize);
+}
+
+//TileLayer.cpp
+WorldMap::TileLayer::Tile::Tile(const sf::Vector2i& position, const TileSheet& tileSheet, int tileID)
+	: m_position(position),
+	m_tileID(tileID),
+	m_tileSheet(tileSheet)
+{
+	m_sprite.setTexture(m_tileSheet.getTexture());
+	m_sprite.setTextureRect(m_tileSheet.getTileLocationByID(m_tileID));
+	const int tileSize = m_tileSheet.getDetails().m_tileSize;
+	m_sprite.setPosition(sf::Vector2f(m_position.x * tileSize, m_position.y * tileSize));
+}
+
+WorldMap::TileLayer::Tile::Tile(const Tile& orig)
+	: m_position(orig.m_position),
+	m_tileID(orig.m_tileID),
+	m_tileSheet(orig.m_tileSheet)
+{
+	m_sprite.setTexture(m_tileSheet.getTexture());
+	m_sprite.setTextureRect(m_tileSheet.getTileLocationByID(m_tileID));
+	const int tileSize = m_tileSheet.getDetails().m_tileSize;
+	m_sprite.setPosition(sf::Vector2f(m_position.x * tileSize, m_position.y * tileSize));
+}
+
+WorldMap::TileLayer::TileLayer(const std::vector<std::vector<int>>& tileMapData, const sf::Vector2i& mapSize, const TileSheet& tileSheet, std::string&& name)
+	: m_name(std::move(name)),
+	m_tileMap()
+{
+	loadInTileMap(tileMapData, mapSize, tileSheet);
+}
+
+void WorldMap::TileLayer::loadInTileMap(const std::vector<std::vector<int>>& tileMapData, const sf::Vector2i& mapSize, const TileSheet& tileSheet)
+{
+	if (!m_tileMap.empty())
+	{
+		clearMap();
+	}
+
+	m_tileMap.reserve(static_cast<size_t>(mapSize.x * mapSize.y));
+
+	for (int col = 0; col < mapSize.y; ++col)
+	{
+		for (int row = 0; row < mapSize.x; ++row)
+		{
+			int tileDrawID = tileMapData[col][row]; //Get ID for tile
+			if (tileDrawID > 0)
+			{
+				m_tileMap.emplace_back(sf::Vector2i(row, col), tileSheet, tileDrawID);
+			}
+		}
+	}
 }
 
-void WorldMap::assignTileLayer(const std::vector<std::vector<int>>& tileData, const sf::Vector2i & mapSize, std::string&& tileLayerName)
+void WorldMap::TileLayer::draw(sf::RenderWindow& window)
 {
-	if (!hasTileLayer(tileLayerName))
+	for (const auto& tile : m_tileMap)
 	{
-		m_tileLayers.emplace_back(tileData, mapSize, m_tileSheetDetails, std::move(tileLayerName), m_mapTexture);
+		window.draw(tile.m_sprite);
 	}
 }
 
+void WorldMap::TileLayer::clearMap()
+{
+	m_tileMap.clear();
+}
+
+const std::string& WorldMap::TileLayer::getName() const
+{
+	return m_name;
+}
+
+//WorldMap.cpp
```

Comment:
These comments were maybe helpful during re-organization, but they make no sense now.  You can still have a comment like this (I do in cpp files with multiple logical segments), but remove the '.cpp' part.